### PR TITLE
[oneDNN EP] Optimized DynamicQuantizeLinear operator

### PR DESCRIPTION
**Pull request by eralmual**
* Removed unnecesary reorders
* Removed unnecesary element wise clip

**Description**: Describe your changes.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/core/providers/dnnl/subgraph/dnnl_dynamicquantizelinear.cc [hunk 1]
```diff
@@ -17,137 +17,126 @@ Y_ZeroPoint = np.clip(round((0 - x_min) / Y_Scale), 0, 255).astype(np.uint8)
...
-  //compute min first since max_reduction needs min dst as post op arg
-  // compute x min
-  sp.AddPrimitive(min_reduction_prim, min_reduction_args);
-  // compute y scale f32
-  sp.AddPrimitive(max_reduction_prim, max_reduction_args);
+  // y_zero_point = np.clip(round((0 - x_min) / y_scale), 0, 255).astype(np.uint8)
+  // Y_ZERO_POINT COMPUTATION 
+  // Create memory and primitive descriptors
+  auto y_zp_md = dnnl::memory::desc(one_dim, dnnl::memory::data_type::u8, x_format);
+  auto zp_prim_d = dnnl::binary::desc(dnnl::algorithm::binary_div, y_scale_md, y_scale_md, y_zp_md);
```
Comment (georgen117) [inline onnxruntime/core/providers/dnnl/subgraph/dnnl_dynamicquantizelinear.cc:87]: am i reading this correctly you are doing an in place binary div.  i.e. y_scale = y_scale/y_zp?

**Review by georgen117 — APPROVED** [review]
Every thing looks really good to me. I would suggest trying to resolve the cpplint warings that are all just whitespace warning.

File: onnxruntime/core/providers/dnnl/subgraph/dnnl_dynamicquantizelinear.cc (hunk 1 above)
Comment (eralmual) [reply onnxruntime/core/providers/dnnl/subgraph/dnnl_dynamicquantizelinear.cc:87]: No, this part means that src0 and src1 have the same md (y_scale_md) and the output md is y_zp_md

Comment (georgen117) [reply onnxruntime/core/providers/dnnl/subgraph/dnnl_dynamicquantizelinear.cc:87]: 🤦‍♂️ i must have been over thinking it yesterday. I feel kinda silly that I even asked.  Thanks :)

**Pull request closed without merging**